module/vision/BallDetector/src/BallDetector.cpp

```diff
@@ -106,137 +108,167 @@ namespace module::vision {
+                const Eigen::Affine3f Hcw(horizon.Hcw.cast<float>());
 
+                // CHECK EACH CLUSTER FOR VALID BALL
                 for (auto& cluster : clusters) {
                     Ball b;
+                    b.covariance = config.ball_angular_cov.asDiagonal();
 
-                    // Average the cluster to get the cones axis
-                    Eigen::Vector3f axis = Eigen::Vector3f::Zero();
+                    // FIND CENTRAL AXIS OF BALL
+                    // Add up all the unit vectors of each point (camera to point in world space) in the cluster to find
+                    // an average vector, which represents the central cone axis
+                    // uBCw: unit vector from camera to ball central axis in world space
+                    Eigen::Vector3f uBCw = Eigen::Vector3f::Zero();
                     for (const auto& idx : cluster) {
-                        axis += rays.col(idx);
+                        uBCw += uPCw.col(idx);
                     }
-                    axis /= cluster.size();
-                    axis.normalize();
+                    uBCw.normalize();  // get cone axis as a unit vector
 
-                    // Find the ray with the greatest distance from the axis
-                    // Should we use the average distance instead?
+                    // FIND UNIT RADIUS OF BALL
+                    // Find the ray (uPCw) with the greatest distance from the central axis (uBCw) to then determine the
+                    // largest radius possible from the edge points available. The radius is the upper bound on the ball
+                    // radius if it were a metre away. arccos(radius) is the angle between the central ball
+                    // axis (uBCw) and the edge of the ball. This helps to find the approximate distance to the ball
                     float radius = 1.0f;
                     for (const auto& idx : cluster) {
-                        const Eigen::Vector3f& ray(rays.col(idx));
-                        if (axis.dot(ray) < radius) {
-                            radius = axis.dot(ray);
-                        }
+                        // Unit vector from the camera to the ball edge, in world space
+                        const Eigen::Vector3f& uECw(uPCw.col(idx));
+                        // Choose this vector if it's further than previously seen vectors, to get the largest angle
+                        radius = uBCw.dot(uECw) < radius ? uBCw.dot(uECw) : radius;
                     }
 
-                    // Ball cam space info
-                    b.cone.axis   = horizon.Hcw.topLeftCorner<3, 3>().cast<float>() * axis;
-                    b.cone.radius = radius;
+                    // The vectors are in world space, multiply by Rcw to get the central axis in camera space
+                    b.uBCc   = Hcw.rotation() * uBCw;
+                    b.radius = radius;  // arccos(radius) is the angle between the central axis and edge of the ball
```

Comment:
Probably better to refer to `radius` being equal to `cos(theta)`, where `theta` is the angle between the central axis and edge of the ball